murmur128.go

```diff
@@ -26,30 +26,30 @@ type Hash128 interface {
 	Sum128() (uint64, uint64)
 }
 
-// digest128 represents a partial evaluation of a 128 bites hash.
-type digest128 struct {
+// Digest128 represents a partial evaluation of a 128 bites hash.
+type Digest128 struct {
 	digest
 	h1 uint64 // Unfinalized running hash part 1.
 	h2 uint64 // Unfinalized running hash part 2.
 }
 
 // New128 returns a 128-bit hasher
-func New128() Hash128 { return New128WithSeed(0) }
+func New128() *Digest128 { return New128WithSeed(0) }
 
 // New128WithSeed returns a 128-bit hasher set with explicit seed value
-func New128WithSeed(seed uint32) Hash128 {
-	d := new(digest128)
+func New128WithSeed(seed uint32) *Digest128 {
```

Comment:
This function is too complex to be inlined (at least in Go 1.17):
```
./murmur128.go:40:6: cannot inline New128WithSeed: function too complex: cost 93 exceeds budget 80
```
Do you think we could get the complexity down to allow inlining and help escape analysis place those values on the stack when possible?